src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -261,6 +261,37 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!--
     ============================================================
+    ProjectReference Targets: Include transitive project references before 
+                              ResolveProjectReferences is called
+    - _ComputeTransitiveProjectReferences
+    - IncludeTransitiveProjectReferences
+    ============================================================
+    -->
+
+  <Target Name="_ComputeTransitiveProjectReferences"
+          DependsOnTargets="RunResolvePackageDependencies"
+          Returns="_TransitiveProjectReferences">
+    <ItemGroup>
+      <ProjectDefs Include="@(PackageDefinitions->WithMetadataValue('Type', 'project'))" />
+
+      <_TransitiveProjectReferences Include="%(ProjectDefs.Path)">
+        <ResolvedPath>%(ProjectDefs.ResolvedPath)</ResolvedPath>
+      </_TransitiveProjectReferences>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="IncludeTransitiveProjectReferences"
+          Condition="'$(ResolvePackageDependenciesForBuild)' == 'true' and '$(EnableResolvePackageDependencies)' == 'true' and Exists('$(ProjectAssetsFile)')"
+          DependsOnTargets="_ComputeTransitiveProjectReferences" 
+          BeforeTargets="AssignProjectConfiguration">
```

Comment:
Could we add `IncludeTransitiveProjectReferences` to `ResolvePackageDependenciesForBuildDependsOn`, and then put `BeforeTargets="AssignProjectConfiguration"` on the `ResolvePackageDependenciesForBuild` target?  It seems like that would prevent us from having to repeat the same condition twice.

In any case, I would like to see comments in the targets file itself explaining how the ordering works.  IIUC, the key is that the `AssignProjectConfiguration` is the entry point that reads `ProjectReference` items, so this target needs to run before that, but only if there is an assets file.

Separately, what is the difference between the `ResolvePackageDependenciesForBuild` and `EnableResolvePackageDependencies` properties?  They seem to only be used together.